src/notebooks/controllers/vscodeNotebookController.ts

```diff
@@ -169,39 +163,39 @@ export class VSCodeNotebookController implements Disposable, IVSCodeNotebookCont
         private readonly browser: IBrowserService,
         private readonly extensionChecker: IPythonExtensionChecker,
         private serviceContainer: IServiceContainer,
-        private readonly serverUriStorage: IJupyterServerUriStorage,
-        private readonly platform: IPlatformService
+        private readonly displayDataProvider: ConnectionDisplayDataProvider
     ) {
         disposableRegistry.push(this);
+        displayDataProvider.onDidChange(
+            (e) => {
+                if (e.connectionId === this.connection.id) {
+                    this.updateDisplayData();
+                }
+            },
+            this,
+            this.disposables
+        );
         this._onNotebookControllerSelected = new EventEmitter<{
             notebook: NotebookDocument;
             controller: VSCodeNotebookController;
         }>();
 
+        const displayData = this.displayDataProvider.getDisplayData(this.connection);
         traceVerbose(
-            `Creating notebook controller for ${kernelConnection.kind} (id=${kernelConnection.id}) with name '${label}'`
+            `Creating notebook controller for ${kernelConnection.kind} (id=${kernelConnection.id}) with name '${displayData.label}'`
         );
         this.controller = this.notebookApi.createNotebookController(
             id,
             _viewType,
-            label,
+            displayData.label,
             this.handleExecution.bind(this),
             this.getRendererScripts(),
             []
         );
+        this.updateDisplayData();
 
         // Fill in extended info for our controller
         this.controller.interruptHandler = this.handleInterrupt.bind(this);
-        this.controller.description = getKernelConnectionDisplayPath(kernelConnection, this.workspace, this.platform);
-        this.controller.detail =
```

Comment:
All of the display information, labels, details, description etc have been moved into a seprate class `DisplayProivder`.